circuitbreakerpanel/circuitbreakerpanel-java-mvn/circuitbreakerpanel-impl/src/main/java/com/lightbend/lagom/recipes/cbpanel/impl/DemoServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package com.lightbend.lagom.recipes.cbpanel.impl;
+
+import akka.NotUsed;
+import com.lightbend.lagom.javadsl.api.ServiceCall;
+import com.lightbend.lagom.javadsl.client.CircuitBreakersPanel;
+import com.lightbend.lagom.recipes.cbpanel.api.DemoService;
+import com.lightbend.lagom.recipes.cbpanel.impl.repository.Repository;
+
+import javax.inject.Inject;
+import java.util.concurrent.CompletableFuture;
+
+public class DemoServiceImpl implements DemoService {
+    
+    private Repository repository;
+    
+    private CircuitBreakersPanel circuitBreaker;
+    
+    @Inject
+    public DemoServiceImpl(Repository repository, CircuitBreakersPanel circuitBreaker) {
+        this.repository = repository;
+        this.circuitBreaker = circuitBreaker;
+    }
+    
+    @Override
+    public ServiceCall<NotUsed, String> getHelloOrFail(String userName) {
+        return request -> circuitBreaker.withCircuitBreaker("breakerA",
```

Comment:
This _is_ the purpose of the recipe. Please, make sure there's comments in the code itself and a link from the `README` file.